### ISSUE TITLE
Support more method signatures in WeakEventSource<TEventArgs>

### ISSUE THREAD

**Issue by Silic0nS0ldier** [tibel/Weakly#29]
Currently `WeakEventSource<TEventArgs>` assumes every delegate will conform to `EventHandler<TEventArgs>` which while _generally_ is a reasonable assumption runs into issues if this isn't true.

In my case, the method signature is `PropertyChangedEventHandler` (to satisfy the `INotifyPropertyChanged` interface) which as far as C#'s type system is concerned, is incompatible with `EventHandler<PropertyChangedEventArgs>`.

There are ways to work around this limitation, but ideally Weakly would support event listeners with flexible arguments. As for _how_ this could be achieved, an idea is to have an anonymous function passed to `WeakEventSource<TEventArgs>.Raise` which accepts a listener and calls it with the intended arguments. Not quite as clean as the existing API, but deals with the technical challenge of handling a variable number of arguments of various types.

Thoughts?

**Comment by tibel** [comment]
I have published a new version of _Weakly 2.9.0_.

Included is class `WeakEventSourceBase<TEventHandler>` to support such scenarios.

```c#
public sealed class PropertyChangedWeakEventSource : WeakEventSourceBase<PropertyChangedEventHandler>
{
    public void Raise(object sender, PropertyChangedEventArgs e)
    {
        var handlers = GetHandlers();

        for (var i = 0; i < handlers.Count; i++)
        {
            handlers[i].Invoke(sender, e);
        }
    }
}
```

**Comment by Silic0nS0ldier** [comment]
Addresses my use case, thanks. Closing.
